addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -159,7 +159,11 @@ function factory(dependencies) {
          * @override
          */
         _updateAfter(previous) {
-            if (this.thread && this.thread !== previous.thread) {
+            if (
+                this.shouldLoadMessages &&
+                this.thread &&
+                (this.thread !== previous.thread || !previous.shouldLoadMessages)
+            ) {
```

Comment:
Note sure if out of scope or a good opportunity, but this logic should be moved into `threadCache`. That would be a first step to get rid of this `_updateAfter`.

Reply:
threadCache is created by the threadViewer or by the thread ?

Reply:
Check `cache` method on thread and who calls it

Reply:
The main cache is created by thread itself. So if you move that logic to thread cache, as soon as thread would be created, some messages will be loaded ... So I think it's not that easy, you'll need something to prevent the threadcache to load messages on creation if you move that logic.

Reply:
Yes this is the goal of the PR. The idea is to fix it by loading data only if there are threadViewers linked to the thread cache.

Reply:
Moved it to a computed field :+1:
```js
/**
 * @private
 */
_computeShouldLoadMessages() {
    if (!this.thread || this.thread.viewers.length === 0) {
        return false;
    }
    if (this.isLoaded || this.isLoading) {
        return false;
    }
    this.loadMessages();
    return true;
}
```

This seems to be working fine. But I still need to fire the `addComponentHint('change-of-thread-cache')` and handle the loading in viewers. How can I replace the change of thread cache as in the new thread cache I won't be aware of a change... should it be replaced by a "thread-cache-loaded" component hint (that would be triggered in `handleThreadCacheLoaded`) and the message list processing of the hint would act the same way ?

Reply:
`change-of-thread-cache` could be triggered in an "on change" compute in thread viewer with thread cache as dependency?